app/src/main/java/com/willowtree/vocable/presets/RoomPresetCategoriesRepository.kt

```diff
@@ -1,57 +1,82 @@
 package com.willowtree.vocable.presets
 
-import android.content.Context
-import com.willowtree.vocable.room.PresetCategoryHidden
 import com.willowtree.vocable.room.CategorySortOrder
 import com.willowtree.vocable.room.PresetCategoryDto
+import com.willowtree.vocable.room.PresetCategoryHidden
 import com.willowtree.vocable.room.VocableDatabase
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.onStart
 import kotlinx.coroutines.sync.Mutex
 import kotlinx.coroutines.sync.withLock
 
 class RoomPresetCategoriesRepository(
-    context: Context
+    private val database: VocableDatabase
 ) : PresetCategoriesRepository {
     // Needed because we're checking database contents and making updates based
     // on them. Multiple callers may call at once.
     private val categoryMutex = Mutex()
-    private val database = VocableDatabase.getVocableDatabase(context)
 
-    override suspend fun getPresetCategories(): List<Category.PresetCategory> {
-        return categoryMutex.withLock {
-            val presetDtos = database.presetCategoryDao().getPresetCategoryDtos()
+    override fun getPresetCategories(): Flow<List<Category.PresetCategory>> {
+        return database.presetCategoryDao().getAllPresetCategoriesFlow().map {
+            it.map {
+                Category.PresetCategory(
+                    it.categoryId,
+                    it.sortOrder,
+                    it.hidden,
+                    PresetCategories.values()
+                        .first { presetCategory -> presetCategory.id == it.categoryId }
+                        .getNameId()
+                )
+            }
+        }.onStart { ensurePopulated() }
```

Comment:
nit: lots of nested `it`s

Reply:
Good feedback. This is super interesting, because copilot wrote this, and I didn't do a good enough job reviewing it to see that!

Reply:
Writing it by hand, I think it would've stood out to me more